File: components/management/src/main/java/org/cloudfoundry/credhub/Management.java
package org.cloudfoundry.credhub;

public class Management {
  private boolean readOnlyMode;

  public Management() {
    super();
    // no arg constructor required by Jackson
  }

  public Management(final Boolean readOnlyMode) {
    super();
    this.readOnlyMode = readOnlyMode;
  }

  public boolean isReadOnlyMode() {
    return readOnlyMode;
  }

  public void setReadOnlyMode(final boolean readOnlyMode) {
    this.readOnlyMode = readOnlyMode;
  }

  @Override
  public String toString() {
    return "isReadOnly: " + isReadOnlyMode();
  }
}

File: components/management/src/main/kotlin/org/cloudfoundry/credhub/Management.kt
package org.cloudfoundry.credhub

class Management {
    var readOnlyMode: Boolean = false

    constructor() : super() {
        // no arg constructor required by Jackson
    }

    constructor(readOnlyMode: Boolean?) : super() {
        this.readOnlyMode = readOnlyMode!!
    }

    override fun toString(): String {
        return "isReadOnly: $readOnlyMode"
    }
}
